File: terre03.rb
# L'alphabet à partir de

argument = ARGV                                      # variable contenant la lettre entrée dans le terminal par l'utilisateur

lettre = ARGV[0].downcase                            # met la lettre en minuscule


if argument.length != 1 || ARGV[0].length != 1 || lettre =~ /[^a-z]/      # si l'utilisateur n'entre rien ou + d'un argument, ou un argument avec plus d'une lettre, ou autre qu'une lettre
    puts "erreur : n'insérer qu'une lettre"                               # affiche ce message d'erreur
else
    print lettre                    
    until lettre == "z"                              # jusqu'à ce que la lettre suivante affichée soit z
        lettre = lettre.succ                         # à chaque tour de la boucle on remplacera le contenu de la variable lettre par la lettre suivante de l'alphabet
        print lettre                                 # à chaque tour de la boucle on affiche la lettre suivante
    end  
end

puts "\n"

File: terre03.js
// Alphabet à partir de

let arguments = process.argv;

let letter = process.argv[2];

let alphabet = "";

if (!letter) {
  console.log("erreur : insérez une lettre");
} else if (letter.length > 1) {
  console.log("erreur : n'insérez qu'une lettre");
} else if (!/^[a-zA-Z]+$/.test(letter)) {
  console.log("erreur : n'insérez rien d'autre qu'une lettre");
} else if (/^[a-z]+$/.test(letter)) {
  let codeAscii = letter.charCodeAt(0);
  for (; codeAscii < 123; codeAscii++) {
    alphabet += String.fromCharCode(codeAscii);
  }
  console.log(alphabet);
} else if (/^[A-Z]+$/.test(letter)) {
  let codeAscii = letter.charCodeAt(0);
  for (; codeAscii < 91; codeAscii++) {
    alphabet += String.fromCharCode(codeAscii);
  }
  console.log(alphabet);
}
